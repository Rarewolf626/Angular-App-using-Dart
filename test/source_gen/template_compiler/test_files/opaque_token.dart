import 'package:angular2/angular2.dart';

const baseUrlToken = const OpaqueToken('baseUrlDescription');

@Component(
  selector: 'has-opaque-tokens',
  template: '{{baseUrl}}',
  providers: const [
    const Provider(baseUrlToken, useValue: 'https://localhost'),
  ],
)
class HasOpaqueTokens {
  final String baseUrl;

  HasOpaqueTokens(@Inject(baseUrlToken) this.baseUrl);
}
