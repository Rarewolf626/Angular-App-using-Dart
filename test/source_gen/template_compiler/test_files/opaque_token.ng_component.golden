[
  {
    "class": "NormalizedComponentWithViewDirectives",
    "component": {
      "class": "Directive",
      "isComponent": true,
      "selector": "has-opaque-tokens",
      "exportAs": null,
      "type": {
        "class": "Type",
        "name": "HasOpaqueTokens",
        "moduleUrl": "asset:angular2/test/source_gen/template_compiler/test_files/opaque_token.dart",
        "prefix": null,
        "isHost": false,
        "value": null,
        "diDeps": [
          {
            "token": {
              "value": null,
              "identifier": {
                "class": "Identifier",
                "name": "baseUrlToken",
                "moduleUrl": "asset:angular2/test/source_gen/template_compiler/test_files/opaque_token.dart",
                "prefix": null,
                "value": null
              },
              "identifierIsInstance": false
            },
            "value": null,
            "isAttribute": false,
            "isSelf": false,
            "isHost": false,
            "isSkipSelf": false,
            "isOptional": false,
            "isValue": false
          }
        ]
      },
      "changeDetection": 6,
      "inputs": {},
      "inputTypes": {},
      "outputs": {},
      "hostListeners": {},
      "hostProperties": {},
      "hostAttributes": {},
      "lifecycleHooks": [],
      "template": {
        "encapsulation": 0,
        "template": "{{baseUrl}}",
        "templateUrl": null,
        "styles": [],
        "styleUrls": [],
        "ngContentSelectors": []
      },
      "providers": [
        {
          "class": "Provider",
          "token": {
            "value": "const OpaqueToken('baseUrlDescription')",
            "identifier": null,
            "identifierIsInstance": false
          },
          "useClass": null,
          "useExisting": null,
          "useValue": {
            "value": "https://localhost",
            "identifier": null,
            "identifierIsInstance": false
          },
          "useFactory": null,
          "useProperty": null,
          "multi": false,
          "deps": null
        }
      ],
      "viewProviders": [],
      "queries": [],
      "viewQueries": []
    },
    "directives": [],
    "pipes": []
  }
]